src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java

```diff
@@ -734,6 +734,7 @@ public void processRerouteFlushRequest(MessageSender sender, String correlationI
         FlowThrottlingData flowThrottlingData = getFlowThrottlingDataBuilder(flow.orElse(null))
                 .correlationId(correlationId)
                 .reason(request.getReason())
+                .flowType(request.getFlowType())
```

Comment:
This method is getting the Flow object from the flowRepository. If you want to reroute a YFlow, the flowtype will be included, but the yFlow will never be found in the flowRepository.

Reply:
This is the builder and it doesn't fetch from the repository. Default type is FLOW and on line 737 it is defined as per request. Request to the database is done on the next stage, when this command is handled - RerouteQueueService::processManualFlushRequest. 

There we do  getFlowInfo(flowId, throttlingData.getFlowType());
and inside you can find:

switch (type) {
            case Y_FLOW:

Reply:
That's right, thanks